runtime/interpreter/interpreter.go

```diff
@@ -4758,7 +4767,7 @@ func (interpreter *Interpreter) invalidateResource(value Value) {
 // UseMemory delegates the memory usage to the interpreter's memory gauge, if any.
 //
 func (interpreter *Interpreter) UseMemory(usage common.MemoryUsage) {
-	if interpreter.memoryGauge == nil {
+	if interpreter == nil || interpreter.memoryGauge == nil {
```

Comment:
I don't think interpreter can be `nil` for any valid cases. If it is, then probably a bug / wrong usage of the method, and might be OK to crash. Was there any particular scenario this was `nil`?

Reply:
This turned out to be due to a test using `nil` instead of an interpreter. AFAICT there's no reason to omit the interpreter there so I added it back.